File: 프로그래머스/unrated/135808. 과일 장수/과일 장수.js
// score를 sort(내림차순으로) -> score에서 계속 m개씩 제거하고, 마지막 'm개째 요소'*m*1를 answer에 더해가기
// 언제까지? -> m개를 채울 수 없을때까지

// function solution (k, m, score) {
//   let answer = 0
//   const sortedScore = score.sort((a,b) => b - a)

//   while(sortedScore.length >= m) {
//     answer += (sortedScore.slice(0,m)[m-1]) * m
//     sortedScore.splice(0,m)  
//   }
//   return answer
// } //-> 실패
function solution (k, m, score) {
  let answer = 0
  const sortedScore = score.sort((a,b) => b - a)
  const boxArr = []

  for(let i = 0 ; i < score.length ; i += m) {
    if( sortedScore.slice(i, i+m).length >= m ) {
      boxArr.push(sortedScore.slice(i, i+m))
    }
  }
  boxArr.map(v => {
    answer += v[m-1] * m
  })
  return answer
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
# 사과는 상자단위로만 판매하며, 상자를 꽉 채우지 못하면 버리기
# 이익이 발생하지 않는 경우 -> 0
# score를 정렬하고, 상자안에 안 들어가는 낮은 점수대 사과는 버리기

def solution(k, m, score):
    result = 0

    score.sort()
    score = score[len(score)%m:] # 상자안에 안 들어가는 저품질 사과는 버리기

    for k in range(0,len(score),m): # m=4 -> 0,4,8 ...
        result += score[k]*m

    return result

print(solution(4,3,[4, 1, 2, 2, 4, 4, 4, 4, 1, 2, 4, 2]))